### ISSUE TITLE
Display list count 

### ISSUE THREAD

**Issue by thieflord06** [ClearskyApp06/ClearskyUI#188]
Display list count within the list tab view for each list. This gives context to how many users are on the list that a specific user is also on.

Placement guide:

Red: for ones without description
Green: for ones with description

else: a new spot will need to be created to display the count

![image](https://github.com/user-attachments/assets/a7d860c1-e09c-4c62-9c69-6e75f6203dd3)

endpoint: /api/v1/anon/get-list/specific/total/<URI/URL>

**Comment by tb38r** [comment]
Hi @thieflord06 , could you assign this to me?

**Comment by tb38r** [comment]
Hi @thieflord06 , any suggestions on a script changes/package installation I might need to make to allow for hot-reloading like feature? 
As is, I need to refresh the client/restart the server to see changes which is quite time consuming. 
Doesn't seem to a dev script with package.json, possible I'm missing something obvious, any pointers?

**Comment by thieflord06** [comment]
When I was making changes, just a hard refresh on the client worked. It should say something like "waiting for changes" on the server. 

@noahm do you have an answer for this?

**Comment by thieflord06** [comment]
That's something we can look into adding. I'm not a UI guy so I'll have to do some research.

**Comment by noahm** [comment]
Yes, for now you need to manually reload. hot module replacement and live reload is enabled in #175 and I may have that ready by tomorrow.

**Comment by tb38r** [comment]
Hi @thieflord06 , to get the user count for each list, do I use the 'useSingleBlocklistCount' hook with each entry's DID, the 'useBlocklistCount' hook or something else entirely?

**Comment by thieflord06** [comment]
From what my understanding is, each tab has it's own "total" function that you would pass the did to get the count.

listsTotal
useBlocklistCount
useSingleBlocklistCount

**Comment by noahm** [comment]
I don't believe the app currently has a hook to return the total size of a given list. You'll need to add that as part of this task.

**Comment by thieflord06** [comment]
Ah right, so, you would have to pass each list URL to: /api/v1/anon/get-list/specific/total/<URI/URL>

**Comment by tb38r** [comment]
I'm regrettable going to have to un-assign myself from this ticket. Not overly familiar with the api and the how to resolve this specific issue.

**Comment by noahm** [comment]
@tb38r I'd be happy to start a branch for this with a new working hook to access the new API endpoint and let you take over from there if that would be helpful

@thieflord06 I'm trying to test out the API and I'm not getting what look like workable results yet. i.e. would you expect this to return a working response? https://api.clearsky.services/api/v1/anon/get-list/specific/total/https%3A%2F%2Fbsky.app%2Fprofile%2Fdid%3Aplc%3Ajuletk33ic5x2pvhg4rozxhc%2Flists%2F3lc7rm6n7yg2m
(just added a URI-encoded version of this list's URL: https://bsky.app/profile/did:plc:juletk33ic5x2pvhg4rozxhc/lists/3lc7rm6n7yg2m)

**Comment by tb38r** [comment]
@noahm that'd be great, thanks

**Comment by thieflord06** [comment]
@noahm I just fixed the issue. Staging is updated and pushing to prod now.

**Comment by thieflord06** [comment]
Staging has the most recent logic but, for a given URL/URI you will either get a count=int, count=0, 400 response, 404 response.

If the total users on the given list can't be found in the db (for a valid URL/URI) it will return a count=0.

**We are still working to get all lists and listitems (users on that list) so there may be valid lists that return 0 until this is complete.

**Comment by tb38r** [comment]
@thieflord06 , I've tested the endpoint on postman with a few different list urls and it works and returns a count consistently as intended. 

Will wait for Noah to create the new hook as mentioned.

Was wondering, could the AccountListEntry type (and the corresponding logic that fetches data for it) be amended to account for a count key so getList() returns more granular info to render?

**Comment by noahm** [comment]
> could the AccountListEntry type (and the corresponding logic that fetches data for it) be amended to account for a count key so getList() returns more granular info to render
Yes, that's certainly possible, but up to Justin whether or not that's something the database can easily handle.

Just pushed a new `list-counts` branch to get you started: https://github.com/ClearskyApp06/ClearskyUI/compare/list-counts

Note that using this in the most naive way will absolutely hit rate limits from our backend. There needs to be both some client-side throttling added, and we probably should also virtualize the scrolling list view, so that only a subset of all the lists we've fetched are actually rendered (thus limiting how many list sizes we look up from the UI)

**Comment by thieflord06** [comment]
return will be changing from "list_uri" to "list" because it can take either a URL or URI.

**Comment by thieflord06** [comment]
@tb38r Any updates on this?
